projects/rocprofiler-sdk/source/lib/python/rocpd/source/perfetto.cpp

```diff
@@ -389,12 +403,18 @@ write_perfetto(
                         _operation = _extdata.operation.value();
                 }
 
-                auto _category = ::perfetto::DynamicCategory{get_category_string(itr.category)};
+                const char* category_name = get_category_string(itr.category);
+
+                auto _category = std::strcmp(category_name,
+                                             sdk::perfetto_category<sdk::category::none>::name) == 0
+                                     ? ::perfetto::DynamicCategory{itr.category}
+                                     : ::perfetto::DynamicCategory{category_name};
+
                 TRACE_EVENT_BEGIN(_category,
                                   ::perfetto::DynamicString{_name},
                                   track,
                                   itr.start,
-                                  ::perfetto::Flow::Global(itr.stack_id ^ uuid_pid),
+                                  ::perfetto::Flow::Global(++global_flow_index),
```

Comment:
```suggestion
                                  ::perfetto::Flow::Global((itr.stack_id == 0) ? (++global_flow_index) : (itr.stack_id ^ this_pid_track.uuid)),
```
It's not pretty, but I change it to this, so that IF a proper stack_id is provided, it will draw the flow arrows.  
BUT we need to get global_flow_index, to some high initial starting number.  So that it avoids collisions with actual stack_id values.

Reply:
Done.